dds/DCPS/RTPS/Spdp.h

```diff
@@ -393,12 +395,19 @@ class OpenDDS_Rtps_Export Spdp : public DCPS::LocalParticipant<Sedp>
   ACE_Condition_Thread_Mutex shutdown_cond_;
   ACE_Atomic_Op<ACE_Thread_Mutex, bool> shutdown_flag_; // Spdp shutting down
 
-  void remove_expired_participants();
   void get_discovered_participant_ids(DCPS::RepoIdSet& results) const;
 
   BuiltinEndpointSet_t available_builtin_endpoints_;
   Sedp sedp_;
 
+  typedef OPENDDS_MULTIMAP(DCPS::MonotonicTimePoint, DCPS::RepoId) TimeQueue;
+
+  void remove_lease_expiration_i(DiscoveredParticipantIter iter);
+  void update_lease_expiration_i(DiscoveredParticipantIter iter,
+                                 const DCPS::MonotonicTimePoint& now);
+  void process_lease_expirations(const DCPS::MonotonicTimePoint& now);
+  TimeQueue lease_expirations_;
```

Comment:
This TimeQueue thing seems like it could pretty easily be its own utility class with its own logic and unit test. Just sayin'.